examples/amm/src/contract.rs

```diff
@@ -516,6 +525,30 @@ impl AmmContract {
             Operation::RemoveAllAddedLiquidity { owner: _ } => {
                 Err(AmmError::RemovingLiquidityLocally)
             }
+
+            Operation::CloseChain => {
+                for account in self.state.shares.indices().await? {
+                    let current_shares = self.state.current_shares_or_default(&account).await;
+                    let (amount_token0, amount_token1) = self
+                        .get_amounts_from_shares(current_shares)
+                        .expect("Getting amounts from shares shouldn't fail!");
+
+                    self.return_shares(
+                        account,
+                        current_shares,
+                        current_shares,
+                        0,
+                        amount_token0,
+                        amount_token1,
+                    );
+                }
+
+                ensure!(
+                    *self.state.total_shares_supply.get() == Amount::ZERO,
+                    AmmError::UntrackedLiquidity
+                );
+                Ok(())
```

Comment:
You still need to call `runtime.close_chain()`.

Reply:
🤦🏻‍♂️ goot catch

Reply:
Is this really resolved? I don't see it.

Reply:
Sorry, I mark it as resolved when I have altered it locally, for me to keep track of the comments I'm addressing ha I'll update this with all the changes soon